libibverbs/dummy_ops.c

```diff
@@ -369,11 +369,15 @@ static int query_device(struct ibv_context *context,
 	return EOPNOTSUPP;
 }
 
+/* Provide a generic implementation for all providers that don't implement
+ * query_device_ex.
+ */
 static int query_device_ex(struct ibv_context *context,
 			   const struct ibv_query_device_ex_input *input,
 			   struct ibv_device_attr_ex *attr, size_t attr_size)
 {
-	return EOPNOTSUPP;
+	memset(attr, 0, sizeof(*attr));
```

Comment:
This has to be attr_size not sizeof(attr) right?

Reply:
Or maybe it should be like

if (attr_size < sizeof(attr->orig_attr)) return EOPNOTSUPP;
memset(&attr->orig_attr, 0, sizeof(attr->orig_attr))
return ibv_query_device(..)

Reply:
And also it has to check that the incoming comp mask is zero and fail.

Reply:
yes, attr_size and not sizeof. 
attr_size  < sizeof(attr->orig_attr) will happen only if ibv_device_attr will change, but that's why _ex was added, can add the check, but then it should probably be added for all calls in providers that do implement the callback as well?